PD-PiotrKurzydym/praca-domowaJS-nr1-Piotr-Kurzydym.js

```diff
@@ -0,0 +1,605 @@
+      
+
+  */
+
+//  DANE WEJŚCIOWE
+console.log(`Zadanie 3.`);
+const colors = ["red", "green", "yellow", "blue", "pink", "orange"];
+const people3 = [
+  {
+    firstName: "Bartolomeo",
+    lastName: "Lozano",
+    nickname: "Rabona",
+    getFavouriteColor(a) {
+      if (a < 1) {
+        console.log(
+          `Podałeś za małą liczbę, liczba nie może być mniejsza niż 1`
+        );
+      } else if (a > 30) {
+        console.log(
+          `Podałeś za dużą liczbę, liczba nie może być większa niż 30`
+        );
+      } else if (a === undefined) {
+        const sum =
+          this.firstName.length + this.lastName.length + this.nickname.length;
+        const index = Math.abs(sum - 5) % colors.length;
+        const FavouriteColor = colors[index];
+        console.log(`FavouriteColor:`, FavouriteColor);
+      } else {
+        const sum =
+          this.firstName.length + this.lastName.length + this.nickname.length;
+        const index = Math.abs(sum - a) % colors.length;
+        const FavouriteColor = colors[index];
+        console.log(`FavouriteColor:`, FavouriteColor);
+      }
+    },
+  },
+];
+people3[0].getFavouriteColor(5);
+console.log(`---`);
+/*
+      3. 
+      a) Dodaj do każdego obiektu funkcję getFavouriteColor
+      b) funkcja ma przyjmować jeden parametr typu number z zakresu 1 - 30
+      c) jeżeli podany parametr jest poza zakresem, powinien wyświetlić się odpowiedni komunikat
+          - podałeś za małą liczbę, liczba nie może być mniejsza niż 1
+          - podałeś za dużą liczbę, liczba nie może być większa niż 30
+      d) w przypadku wywołania funkcji bez parametru, powinniśmy ustawić domyślną wartość na 5
+      e) funkcja powinna zsumować wszystkie litery imienia, nazwiska i przezwiska, 
+      odjąć od tej sumy liczbę wprowadzoną w parametrze, a następnie za pomocą działania modulo (%) względem długości tablicy kolorów
+      wyznaczyć index
+      f) za pomocą indexu funkcja powinna wyciągnąć odpowiedni kolor z tablicy i wyświetlić go w konsoli.
+      Dla powyższego przykładu i liczby 5 wprowadzonej w parametrze, powinniśmy uzyskać wynik:
+      (22 - 5) % 6 = 5
+      console.log("orange")
+      Hints
+      - jeżeli po odjęciu parametru funkcji od sumy liter uzyskacie wartośc ujemną, możecie użyć metody z biblioteki Math, 
+      Math.abs(-20), która zamieni liczbę na wartość absolutną, czyli dodatnią
+      - w funkcji musicie użyć słówka this, parametru i tablicy, która jest na zewnątrz, tablica z kolorami może mieć
+      dowoloną ilość kolorów
+  */
+
+console.log(`Zadanie 4.a) b)`);
+const colors2 = ["red", "green", "yellow", "blue", "pink", "orange"];
+const people4 = [
+  {
+    firstName: "Bartolomeo",
+    lastName: "Lozano",
+    nickname: "Rabona",
+  },
+  {
+    firstName: "Piotrulo",
+    lastName: "Kurzydymulo",
+    nickname: "Kurzy",
+  },
+  {
+    firstName: "James",
+    lastName: "Hetfield",
+    nickname: "Jaymz",
+  },
+];
+function getFavouriteColor2(a, arrayIndex) {
```

Comment:
W tym zadaniu dobrze widać jaką możliwość dają funkcje. Kod poniżej jest w zasadzie kopią kodu z zadania trzeciego. Oznacza to że śmiało można zamknąć część logiki w funkcji i reużyć  jej w innej funkcji.